Application/EdFi.Ods.Common/Models/Validation/IsApiVersionValidValidator.cs

```diff
@@ -0,0 +1,47 @@
+﻿// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using EdFi.Ods.Common.Models.Definitions;
+using FluentValidation;
+using Newtonsoft.Json;
+
+namespace EdFi.Ods.Common.Models.Validation
+{
+    public class IsApiVersionValidValidator : AbstractValidator<string>
+    {
+        private readonly string _apiVersion;
+
+        public IsApiVersionValidValidator(string apiVersion)
+        {
+            var regex = new Regex(@"ApiModel.*.json");
+
+            _apiVersion = apiVersion;
+
+            RuleFor(extensionFolder => extensionFolder)
+                .Must(
+                    e =>
+                    {
+                        var metaDataFolder = Path.Combine(e, "Artifacts", "Metadata");

```

Comment:
we probably need some logging here. I think the intent is that we have an extension and that the model is valid. should be testing both of these rules as  compound?

Reply:
We are testing both conditions and logging is done in AssemblyLoaderHelper.

Reply:
Actually, I'm not checking both conditions, only validating version here.  There's another validator that checks for extensions.

Reply:
I'm not sure whey we need to make a compound validator here.  Both IsApiVersionValidValidator and IsExtensionPluginValidator are called from AssemblyLoaderHelper.  Combining these two validators into two rules in one validator will only check for valid extension assemblies.  Making the validators more granular allows us to check if the version is valid for extension, profiles and composites, right?